nemo/collections/common/tokenizers/bytelevel_tokenizers.py

```diff
@@ -38,9 +36,20 @@ def normalize(self, text) -> str:
 
 
 class ByteLevelTokenizer(TokenizerSpec):
-    def __init__(self):
+    def __init__(self, special_tokens: Optional[Union[Dict[str, str], List[str]]] = None):
         self.vocab_size = 259
-        self.special_tokens = [self.bos_id, self.eos_id, self.pad_id]
+        self.special_start = 256
+        self.special_token_to_id = {
+            self.pad_id: self.pad_id,
+            self.bos_id: self.bos_id,
+            self.eos_id: self.eos_id,
+        }
+        special_tokens = {} if special_tokens is None else special_tokens
+        for tok in special_tokens:
```

Comment:
How can special tokens be either dictionary or list?

Reply:
Its a standard thing across tokenizerspec tokenizers that support special tokens (like sentencepiece). Need to support both dict and list to keep usage similar. In case it is dict, the expectation specifically for byte-level is that both key and value will be the same.